### PR TITLE
[ESI] Convert cosim integration test "lit format" to tool

### PR DESCRIPTION
**Pull request by teqdruid**
- Converted 'lit' "format" code to separate tool: `esi-cosim-runner.py` and use lit's default format (ShTest) to run it as separate process.
- Also converted it to use `circt-rtl-sim.py` to compile and run the RTL simulation.
- Code which wasn't stripped out is mostly the same.
- Required paths from build system now configured with cmake when copying to bin dir.

### CODE REVIEW

**Comment by teqdruid** [comment]
Verification integration test build: https://github.com/teqdruid/circt/actions/runs/481719384

**Comment by teqdruid** [comment]
Test integration run: https://github.com/teqdruid/circt/actions/runs/481897957

File: tools/esi/esi-cosim-runner.py.in [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+
+
+def __main__(args):
+    argparser = argparse.ArgumentParser(
+        description="RTL cosimulation runner for ESI")
+    argparser.add_argument("source",
+                           help="The source run spec file")
+    argparser.add_argument("addlArgs", nargs=argparse.REMAINDER,
+                           help="The source run spec file")
```
Comment (mikeurbach) [inline tools/esi/esi-cosim-runner.py.in:224]: I think this help text needs an update?

**Review by mikeurbach — APPROVED** [review]
Nice, looks good to me.

**Comment by teqdruid** [comment]
Thanks for the review @mikeurbach

**Pull request merged**